### PR TITLE
Remove dependency on OpenZeppelin

### PR DESCRIPTION
**Pull request by alcueca**
Published in npm as 1.1.0 To keep using OpenZeppelin contracts stick to the 1.0.7 package.

ERC20 has been replaced by our own, to save 1 SSTORE when `transferFrom` with MAX allowance.
Ownable, SafeMath and a few others have just been copied across.

### CODE REVIEW

File: contracts/math/SafeMath.sol [hunk 1]
```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
```
Comment (gakonst) [inline contracts/math/SafeMath.sol:3]: Note if we end up upgrading to 0.8.0 eventually, we won't need this!

File: contracts/token/ERC20.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        return _approve(msg.sender, spender, wad);
+    }
+
+    function transfer(address dst, uint wad) public virtual override returns (bool) {
+        return _transfer(msg.sender, dst, wad);
+    }
+
+    function transferFrom(address src, address dst, uint wad) public virtual override returns (bool) {
+        uint256 allowed = _allowance[src][msg.sender];
+        if (src != msg.sender && allowed != type(uint).max) {
```
Comment (gakonst) [inline contracts/token/ERC20.sol:43]: nice

File: dist/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { BigNumber, BigNumberish } from "ethers";
```
Comment (gakonst) [inline dist/index.d.ts]: Oops, let's not have these files in

**Review by gakonst — APPROVED** [review]

File: contracts/math/SafeMath.sol (hunk 1 above)
Comment (alcueca) [reply contracts/math/SafeMath.sol:3]: Yeah, upgrading to 0.8 is next in line.

**Pull request merged**